tests/python/unittest/test_tvm_testing_features.py

```diff
@@ -210,5 +210,44 @@ def test_pytest_mark_covariant(self, request, target, other_param):
         self.check_marks(request, target)
 
 
+@pytest.mark.skipif(
+    bool(int(os.environ.get("TVM_TEST_DISABLE_CACHE", "0"))),
+    reason="Cannot test cache behavior while caching is disabled",
+)
+class TestCacheableTypes:
+    class EmptyClass:
+        pass
+
+    @tvm.testing.fixture(cache_return_value=True)
+    def uncacheable_fixture(self):
+        return self.EmptyClass()
+
+    @pytest.mark.xfail(reason="Requests cached fixture of uncacheable type", strict=True)
```

Comment:
is there any way to assert on the specific error generated here?

Reply:
Normally, yes, with `pytest.raises`.  Unfortunately, I don't think that's possible here, because the exception is raised during the fixture function definition.

Reply:
seems like you could just invoke uncacheable_fixture from the test body, no?

Reply:
Good point.  It can't be called directly, but `request.getfixturevalue('uncacheable')` does trigger it within the test.  Updated.